src/Surfnet/AzureMfa/Application/Service/AzureMfaService.php

```diff
@@ -19,42 +19,30 @@
 namespace Surfnet\AzureMfa\Application\Service;
 
 use Exception;
-use RobRichards\XMLSecLibs\XMLSecurityKey;
 use SAML2\Assertion;
-use SAML2\Certificate\PrivateKeyLoader;
-use SAML2\Configuration\PrivateKey;
 use Surfnet\SamlBundle\Entity\IdentityProvider;
 use Surfnet\SamlBundle\Entity\ServiceProvider;
 use Surfnet\SamlBundle\Http\PostBinding;
-use Surfnet\SamlBundle\SAML2\AuthnRequest;
 use Surfnet\SamlBundle\SAML2\AuthnRequestFactory;
 use Symfony\Component\HttpFoundation\RedirectResponse;
 use Symfony\Component\HttpFoundation\Request;
 
 class AzureMfaService
 {
-
-    /**
-     * @var string
-     */
-    private $publicKey;
-    /**
-     * @var string
-     */
-    private $privateKey;
     /**
      * @var PostBinding
      */
     private $postBinding;
 
-    public function __construct(
-        PostBinding $postBinding
-    ) {
-        $this->postBinding = $postBinding;
+    /**
+     * @var ServiceProvider
+     */
+    private $serviceProvider;
 
-        // Todo: make keys configurable
-        $this->publicKey = __DIR__ . '/../../../../../vendor/surfnet/stepup-saml-bundle/src/Resources/keys/development_publickey.cer';
-        $this->privateKey = __DIR__ . '/../../../../../vendor/surfnet/stepup-saml-bundle/src/Resources/keys/development_privatekey.pem';
+    public function __construct(ServiceProvider $serviceProvider, PostBinding $postBinding)
```

Comment:
Personally I don't like how this gets implicitly injected through a lot of magic.